### PR TITLE
CI: Test all notebooks

### PR DESCRIPTION
**Pull request by PapyChacal**
This PR updates the outdated Notebooks, and set the CI to inconditionally test them all, with no special treatment.

Prevents surprises like seen in #694.

### CODE REVIEW

**Comment by PapyChacal** [comment]
Oh, right, one notebook needs MLIR, so I still ignore this one specifically in the current CI setup. (It passed locally, for the record!)

**Review by webmiche — APPROVED** [review]

**Comment by webmiche** [comment]
This one should be explicitely covered in the MLIR CI, so that's fine. Maybe add a comment?

File: docs/irdl.ipynb [hunk 1]
```diff
@@ -1200,7 +1200,7 @@
     "    name: str = \"while_op\"\n",
     "    value: Region\n",
     "    \n",
-    "region = Region.from_block_list([Block(arg_types=[i32])])\n",
+    "region = Region.get([Block(arg_types=[i32])])\n",
```
Comment (superlopuh) [inline docs/irdl.ipynb:1203]:
```suggestion
    "region = Region([Block(arg_types=[i32])])\n",
```

File: docs/xdsl-introduction.ipynb [hunk 2]
```diff
@@ -593,13 +593,13 @@
     "printer.print_op(table)\n",
     "\n",
     "# defines a filter region                                     # (1)\n",
-    "filter_region = Region.from_block_list([Block.from_callable(\n",
+    "filter_region = Region.get([Block.from_callable(\n",
```
Comment (superlopuh) [inline docs/xdsl-introduction.ipynb:596]:
```suggestion
    "filter_region = Region([Block.from_callable(\n",
```

**Pull request merged**